src/detect-metadata.h

```diff
@@ -24,8 +24,22 @@
 #ifndef __DETECT_METADATA_H__
 #define __DETECT_METADATA_H__
 
+/**
+ * \brief Signature metadata list.
+ */
+typedef struct DetectMetadata_ {
+    /* pointer to key */
```

Comment:
can you add explanation on who owns these ptrs (hash table as far as I understand)